### PR TITLE
add Cython static analysis

### LINKED ISSUE #29344: static analysis for Cython modules
- [ ] should autogen stubs, or support parsing Cython code directly
- [ ] run on all files/code (no need for function to already be typed like mypy requires)

### PR DESCRIPTION
**Pull request by belimedo**
**Description**

Fixes #29344

In this PR, I added [cython-lint](https://github.com/MarcoGorelli/cython-lint) for static analysis of Cython code. I also run the hook and fixed all of the errors this checker found.

@sshane, @adeebshihadeh If you can, please take a look at this PR, let me know if you wanted something else to be done for this static analysis. I ignored the same directories as `cpplint` checker ignores, and the only contraint I added was the line length (so that it also matches `cpplint` one).

Also, in the current master, file `selfdrive/ui/qt/maps/map_helpers.cc` fails `cppcheck` hook test. I fixed it locally (return statement error), but I would like to include the fix in the repo as well. If you want, I can add the solution to this PR for that file as well.

**Verification**

I run the pre-commit hooks and edited code to pass all of them.

### CODE REVIEW

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
This diff is unreadable. Can you change the indent back to two spaces? Then I should be able to read this.

**Comment by belimedo** [comment]
> This diff is unreadable. Can you change the indent back to two spaces? Then I should be able to read this.

@adeebshihadeh Change to increase indent from 2 to 4 spaces was one of a requirements from `cython-lint`. Changing it back to 2 would make the pre-commit git hook fail. 

If you wish, I can make it back to 2, but that would make git-hook fail and I'd have to use `--no-verify` while committing.

**Comment by adeebshihadeh** [comment]
This need to handle our indent style or we can't use it.

**Comment by belimedo** [comment]
> This need to handle our indent style or we can't use it.

@adeebshihadeh Deal, I will fix it in a second. This linting tool provides us with two flags `--ignore` and `--no-pycodestyle`. The first one allows us to ignore a single rule, for example, indentation (rule E111). The second one will remove all python code checks. It will still do a static analysis as it is giving me an error for included libraries that are not used, but errors for indentation, end of a line with semicolon, 2 empty lines before and after class definitions will be missing. So for now, I will fix the indentation only and re-request the review.

Do you have any other requests regarding static analysis? I asked in the git issue as well, but I didn't get any response from Shane.

File: common/params_pyx.pyx [hunk 1]
```diff
@@ -29,11 +29,13 @@ cdef extern from "common/params.h":
...
 
 def ensure_bytes(v):
-  return v.encode() if isinstance(v, str) else v;
+  return v.encode() if isinstance(v, str) else v
+
 
 class UnknownKeyName(Exception):
   pass
 
+
```
Comment (adeebshihadeh) [inline common/params_pyx.pyx:38]: is this a lint rule? if so, let's disable it and revert these changes

**Review by adeebshihadeh — COMMENTED** [review]
LGTM, let's just remove the new line thing

**Comment by adeebshihadeh** [comment]
> Do you have any other requests regarding static analysis? I asked in the git issue as well, but I didn't get any response from Shane.

We prefer to discuss actual code and PRs, particularly with new contributors.

File: common/params_pyx.pyx (hunk 1 above)
Comment (belimedo) [reply common/params_pyx.pyx:38]: Yep, it is one of the rules. I will revert the change, add new ignore and we are good to go.

**Comment by adeebshihadeh** [comment]
Looks good, thanks!

**Comment by belimedo** [comment]
> 
> We prefer to discuss actual code and PRs, particularly with new contributors.

That sounds reasonable. I guess I'm used to features that are implemented in weeks (sometimes even months), so I engage in a conversation with the issue owner a lot. 

Thanks a lot for your input regarding this PR and issue! You were really helpful, I hope that this is the first of many issues fixed by me on this repo.

**Comment by adeebshihadeh** [comment]
There's lot of open bounties to check out! Bounties are how we organize priorities from external contributors and reward them for contributing.

https://github.com/orgs/commaai/projects/26
